File: DSA/Greedy/Problems/7.Non-overlapping Intervals/sol.java

import java.util.*;

class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[1], b[1]));
        int limit = Integer.MIN_VALUE, count = 0;
        for (int[] interval : intervals) {
            if (interval[0] >= limit) {
                count++;
                limit = interval[1];
            }
        }
        return intervals.length - count;
    }
}

File: DSA/Greedy/Problems/7.Non-overlapping Intervals/sol.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](auto &a, auto &b) {
            return a[1] < b[1];
        });
        int limit = INT_MIN, count = 0;
        for (auto& interval : intervals) {
            if (interval[0] >= limit) {
                count++;
                limit = interval[1];
            }
        }
        return intervals.size() - count;
    }
};
